Replace ESLint with Biome and reformat code

Ubayed-Bin-Sufian opened the pull request:
### Summary

This pull request replaces **ESLint** with **Biome** for improved performance. The configuration is set to prefer single quotes to maintain a cleaner and more efficient code style. Additionally, the code is reformatted using Biome to ensure consistency across the project.

norbusan commented:
That fixes only one line. 
And I believed the problem is that your git checkout might have changed the lines endings.

Ubayed-Bin-Sufian commented:
> That fixes only one line. And I believed the problem is that your git checkout might have changed the lines endings.

Thanks for your kind feedback. 
Could you clarify how my Git checkout might have changed line endings? I’m working on Windows, and I want to make sure my settings don’t cause unexpected changes. I checked my `core.autocrlf` settings, which is set to true.

norbusan commented:
Title of PR is incorrect
Content of the PR is trivial change of quotes.
Can be merged (I am not a maintainer)

Ubayed-Bin-Sufian commented:
> Title of PR is incorrect Content of the PR is trivial change of quotes. Can be merged (I am not a maintainer)

Thanks a lot for your review. I have changed the title as needed.

hongquan commented:
Thanks for attempt, @Ubayed-Bin-Sufian . But in a language where single quote is not different from double quote, I prefer single quote, because:
- It makes the code less busy to eyes (double quote has more strokes).
- It let press less key when typing (double quote require to use Shift + ').

Please change the target to:
- Replace ESLint with [Biome](https://biomejs.dev/), configuring it to prefer single quote.
- Reformat code with Biome.

Ubayed-Bin-Sufian commented:
Thank you so much for your thoughtful feedback, @hongquan! I’ve made the updates as per your suggestions. I would really appreciate it if you could take a look when you have a moment, and let me know if any further adjustments are needed. Thanks again!

hongquan commented:
Sorry, I found you run Biome with `fix --unsafe`. Please apply `safe fix` with Biome only. For issues which Biome consider "unsage", we must fix manually.
@Ubayed-Bin-Sufian

Ubayed-Bin-Sufian commented:
> Sorry, I found you run Biome with `fix --unsafe`. Please apply "safe fix" with Biome only. For issues which Biome consider "unsafe", we must fix manually. @Ubayed-Bin-Sufian

I have dropped the commit with the unsafe fix as it was changing the semantic of the code. Followed the guideline https://biomejs.dev/linter/#safe-fixes. Now Biome can apply automatically without risking breaking changes. @hongquan

hongquan commented:
@Ubayed-Bin-Sufian Could you explain from which source you got the *biome.json*, don't tell me you blindly copied it from some AI answer.

Ubayed-Bin-Sufian commented:
> @Ubayed-Bin-Sufian Could you explain from which source you got the _biome.json_, don't tell me you blindly copied it from some AI answer.

I used this source to create the biome.json file https://biomejs.dev/guides/getting-started/#configuration. As per the instruction we needed to change to double quotes. And for the ignore files, I looked through the eslint.ignore files and added the required ones in the biome.json. I used AI to understand the context of Biome where I didn't understand. For the fix --unsafe part, I blindly copied it without reading the documentation. I apologize for it.

hongquan commented:
I'm Ok with the `--unsafe` mistake, because only someone with some experience with Biome can understand how it could break code.

But I complain about this red part:

![image](https://github.com/user-attachments/assets/1ad7a34e-65f3-46c0-827c-959e994025d0)

If you just follow the example in the documentation https://biomejs.dev/guides/configure-biome/, you won't add it

![image](https://github.com/user-attachments/assets/7ca6b766-f41d-4c5f-bc50-f056a9f9e0d9)

The red part tells Biome to not respect the _.gitignore_, it means that Biome can touch the *node_modules* by accidence!

Ubayed-Bin-Sufian commented:
Thanks a lot for the guidance. I only followed the Getting Started page and creating the biome.json part. I didn't look through the configure Biome section. That's my mistake. I will remove the red highlighted part.

Ubayed-Bin-Sufian commented:
@hongquan Could you please review? 

A short summary of what I have done based on the documentation:

Reference: https://biomejs.dev/guides/configure-biome/

1. Added `include": ["src/**/*.js", "docs/**"]` to explicitly define which files should be processed.
2. Updated `"ignore"` rules to exclude `"docs/images"`, `"**/*.min.js"`, and `"src/icons"` while keeping previous ignored paths.
3. `"vcs"` section was removed as it was not needed for this setup.

I do have few quick questions. 

1. By default indentStyle for fomatter is 'tab'. Do we follow along?
2. Also, linewidth is 80 for formatter. Do we need to change?
3. Do we have any specific linewidth for javascript.formatter?

hongquan commented:
- `Tab` is Ok. This project doesn't mix JS with HTML, so tab is Ok. If mixing, two-whitespace is better.
- Please linewidth increase to 120.
- Please resolve conflict.
